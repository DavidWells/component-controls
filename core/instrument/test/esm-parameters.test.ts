import { loadStoriesTests } from './loadTestFiles';

describe('esm-parameters', () => {
  loadStoriesTests({ stories: { sourceFiles: false } }, ['esm', 'parameters']);
});
