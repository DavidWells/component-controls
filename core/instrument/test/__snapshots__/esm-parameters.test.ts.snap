// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`esm-parameters name-and-parameters.js 1`] = `
Object {
  "components": Object {
    "3880efc294775fe5c4afbb60ef59b162": Object {
      "name": "ControlsTable",
      "package": "9c22f6bdeb80c60a4dcfc1fe822f5f3b",
    },
  },
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {
      "ControlsTable": "3880efc294775fe5c4afbb60ef59b162",
    },
    "date": 2020-07-17T14:45:46.147Z,
    "dateModified": 2020-07-17T14:45:46.148Z,
    "package": "9c22f6bdeb80c60a4dcfc1fe822f5f3b",
    "title": "Story",
  },
  "packages": Object {
    "9c22f6bdeb80c60a4dcfc1fe822f5f3b": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.16.0",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.10.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^25.1.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
      },
      "fileHash": "9c22f6bdeb80c60a4dcfc1fe822f5f3b",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/esm/parameters/name-and-parameters.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.16.0",
    },
  },
  "stories": Object {
    "myStory": Object {
      "arguments": Array [],
      "component": "ControlsTable",
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 31,
          "line": 2,
        },
        "start": Object {
          "column": 23,
          "line": 2,
        },
      },
      "name": "Custom story name",
      "smartControls": Object {
        "smart": false,
      },
      "source": "() => {}",
    },
  },
  "transformed": "export default { title: 'Story' };
export const myStory = () => {};
myStory.story = {
  name: 'Custom story name',
  component: ControlsTable,
  smartControls: {
    smart: false,
  },
};
",
}
`;

exports[`esm-parameters name-parameters-and-controls.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": 2020-07-17T14:45:45.645Z,
    "dateModified": 2020-07-17T14:45:45.646Z,
    "package": "8d5d8b91872154dc5d9f2bae81990c2f",
    "title": "Story",
  },
  "packages": Object {
    "8d5d8b91872154dc5d9f2bae81990c2f": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.16.0",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.10.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^25.1.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
      },
      "fileHash": "8d5d8b91872154dc5d9f2bae81990c2f",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/esm/parameters/name-parameters-and-controls.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.16.0",
    },
  },
  "stories": Object {
    "myStory": Object {
      "arguments": Array [],
      "controls": Object {
        "name": Object {
          "label": "Name",
          "order": 9999,
          "type": "ControlTypes.TEXT",
          "value": "Mark",
        },
      },
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 31,
          "line": 2,
        },
        "start": Object {
          "column": 23,
          "line": 2,
        },
      },
      "name": "Custom story name",
      "smartControls": Object {
        "smart": false,
      },
      "source": "() => {}",
    },
  },
  "transformed": "export default { title: 'Story' };
export const myStory = () => {};
myStory.story = {
  name: 'Custom story name',
  controls: {
    name: {
      type: ControlTypes.TEXT,
      label: 'Name',
      value: 'Mark',
      order: 9999,
    },
  },
  smartControls: {
    smart: false,
  },
};
",
}
`;

exports[`esm-parameters story-property-name.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": 2020-07-17T14:45:46.147Z,
    "dateModified": 2020-07-17T14:45:46.147Z,
    "package": "63ade20bb777cbf8acf363a8059b66af",
    "title": "Story",
  },
  "packages": Object {
    "63ade20bb777cbf8acf363a8059b66af": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.16.0",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.10.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^25.1.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
      },
      "fileHash": "63ade20bb777cbf8acf363a8059b66af",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/esm/parameters/story-property-name.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.16.0",
    },
  },
  "stories": Object {
    "myStory": Object {
      "arguments": Array [],
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 31,
          "line": 2,
        },
        "start": Object {
          "column": 23,
          "line": 2,
        },
      },
      "name": "Custom story name",
      "source": "() => {}",
    },
  },
  "transformed": "export default { title: 'Story' };
export const myStory = () => {};
myStory.story = {
  name: 'Custom story name',
};
",
}
`;
